kyuubi-server/src/main/scala/org/apache/kyuubi/server/trino/api/TrinoContext.scala

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server.trino.api
+
+import java.io.UnsupportedEncodingException
+import java.net.{URLDecoder, URLEncoder}
+import javax.ws.rs.core.{HttpHeaders, Response}
+
+import scala.collection.JavaConverters._
+
+import io.trino.client.{ClientSelectedRole, QueryResults}
+import io.trino.client.ProtocolHeaders.TRINO_HEADERS
+
+// TODO: Request and Respone context, to be enriched
+case class TrinoContext(
+    user: String,
+    timeZone: String,
+    transactionId: String,
```

Comment:
We do not support transaction, should throw exception if it's defined in header.